chat-android/src/main/java/com/ably/chat/Occupancy.kt

```diff
@@ -63,11 +71,34 @@ public fun Occupancy.asFlow(): Flow<OccupancyEvent> = transformCallbackAsFlow {
 }
 
 /**
- * Represents the occupancy of a chat room.
+ * Represents the occupancy event.
  *
  * (CHA-O2)
  */
 public interface OccupancyEvent {
+    /**
+     * Defines the type of the occupancy event, represented by an instance of [OccupancyEventType].
+     */
+    public val type: OccupancyEventType
+
+    /**
+     * Represents data about the occupancy of a chat room, including connection and presence details.
+     *
+     * Provides information such as the number of active connections and the number of presence members in the chat room.
+     */
+    public val occupancy: OccupancyData
+}
+
+public enum class OccupancyEventType {
+    Updated,
+}
```

Comment:
```suggestion
public enum class OccupancyEventType(public val eventName: String) {
    Updated("occupancy.updated"),
}
```
As per https://ably.atlassian.net/wiki/spaces/CHA/pages/3967713291/CHADR-101+Pre-GA+SDK+API+Review#Occupancy

Reply:
Not sure, if this really affects the working of the SDK since we don't serialize it and send over the wire

Reply:
But event enum will be consistent with others